src/jit/decomposelongs.cpp

```diff
@@ -819,6 +822,7 @@ GenTree* DecomposeLongs::DecomposeInd(LIR::Use& use)
     GenTreePtr addrHigh =
         new (m_compiler, GT_LEA) GenTreeAddrMode(TYP_REF, addrBaseHigh, nullptr, 0, genTypeSize(TYP_INT));
     GenTreePtr indHigh = new (m_compiler, GT_IND) GenTreeIndir(GT_IND, TYP_INT, addrHigh, nullptr);
+    indHigh->gtFlags |= indLow->gtFlags & GTF_ALL_EFFECT;
```

Comment:
This should probably copy the `IND`-specific flags as well. I would suggest:

`indHigh->gtFlags |= (indLow->gtFlags & (GTF_GLOB_REF | GTF_EXCEPT | GTF_IND_FLAGS));`